src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -159,31 +174,80 @@ public void actionPerformed(ActionEvent e) {
       }
     });
 
+    //updated to verify password with database as well
     userSignInButton.addActionListener(new ActionListener() {
       @Override
       public void actionPerformed(ActionEvent e) {
         if (userList.getSelectedIndex() != -1) {
           final String data = userList.getSelectedValue();
-          clientContext.user.signInUser(data);
-          userSignedInLabel.setText("Hello " + data);
+          JTextField username = new JTextField(data);
+          JTextField password = new JPasswordField();
+          Object[] message = {
```

Comment:
Having this be a raw Object seems fishy, are you sure you don't want String? 

Also, you're showing the user's password in plain text?